internal/services/gcp_reservation_service.go

```diff
@@ -152,3 +166,12 @@ func CreateGCPReservation(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+func isValidNamePattern(namePattern string) bool {
+	if strings.ToLower(namePattern) != namePattern {
+		return false
+	}
+	// Define a regular expression pattern to match valid RFC-1035-compatible names
+	pattern := regexp.MustCompile(`^[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$`)
+	return pattern.MatchString(namePattern)
+}
```

Comment:
You could have create a package-level unexported variable and compile the expression only once on package import (via `init`), because typically regexp compilation is slower operation than matching by far. But it does not a huge difference in this use case.